### PR TITLE
[UID2-3104] Update Java SDK guide with new interface for DSP check

### PR DESCRIPTION
**Pull request by caroline-ttd**
Changes from https://github.com/IABTechLab/uid2-client-java/pull/63

### CODE REVIEW

File: docs/sdks/uid2-sdk-ref-java.md [hunk 1]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
-## Usage for DSPs
+## Initialization
+
+The initialization step depends on the role, as shown in the following table.
+
+| Role        | Create Instance of Class | Link to Instructions                              |
+|:------------| :--- |:--------------------------------------------------|
+| `DSP`       | BidstreamClient | [Usage for DSPs](#usage-for-dsps)                 |
+| `Sharer`    | SharingClient | [Usage for UID2 Sharers](#usage-for-uid2-sharers) |
+| `Publisher` | Uid2PublisherClient | [Usage for Publishers](#usage-for-publishers)     |
```
Comment (jon8787) [inline docs/sdks/uid2-sdk-ref-java.md:69]: we're missing advertisers here.

File: docs/sdks/uid2-sdk-ref-java.md [hunk 2]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+```java
+client.refresh();
+```
 
-| Function | Description |
-| :--- | :--- |
-| `GetStatus()` | The decryption result status. For a list of possible values and definitions, see [Response Statuses](#response-statuses). |
-| `GetUid()` | The raw UID2 for the corresponding UID2 advertising token. |
-| `GetEstablished()` | The timestamp indicating when a user first established the UID2 with the publisher. |
+3. Decrypt a token into a raw UID2. Pass the domain name of the site where the bid originated from:
```
Comment (sunnywu) [inline docs/sdks/uid2-sdk-ref-java.md:154]: You might want to add [this](https://github.com/IABTechLab/uid2docs/pull/518/files) changes here too

File: docs/sdks/uid2-sdk-ref-java.md (hunk 1 above)
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:69]: Uid2PublisherClient => `PublisherUid2Client`

File: docs/sdks/uid2-sdk-ref-java.md [hunk 3]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
 - [https://central.sonatype.com/artifact/com.uid2/uid2-client](https://central.sonatype.com/artifact/com.uid2/uid2-client)
 
-## Usage for DSPs
+## Initialization
+
+The initialization step depends on the role, as shown in the following table.
+
+| Role        | Create Instance of Class | Link to Instructions                              |
+|:------------| :--- |:--------------------------------------------------|
+| `DSP`       | BidstreamClient | [Usage for DSPs](#usage-for-dsps)                 |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:67]: Class names should be formatted as `code`

File: docs/sdks/uid2-sdk-ref-java.md [hunk 4]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
-The initialization function configures the parameters necessary for the SDK to authenticate with the UID2 service. It also allows you to configure retry intervals in the event of errors.
+You will need to provide the values necessary for the SDK to authenticate with the UID2 service.
 
-| Parameter | Description | Recommended Value |
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the UID2 service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to UID2, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+| Parameter | Description | 
+| :--- | :--- |
+| `endpoint` | The endpoint for the UID2 service. See [Environments](../getting-started/gs-environments) |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:75]: We have a mix of names across the different clients :( This is called `baseUrl` for `BidstreamClient` and `SharingClient` and `uid2BaseUrl` for `PublisherUid2Client`

File: docs/sdks/uid2-sdk-ref-java.md [hunk 5]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+You will need to provide the values necessary for the SDK to authenticate with the UID2 service.
 
-| Parameter | Description | Recommended Value |
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the UID2 service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to UID2, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+| Parameter | Description | 
+| :--- | :--- |
+| `endpoint` | The endpoint for the UID2 service. See [Environments](../getting-started/gs-environments) | 
+| `auth_key` | The API key. See [UID2 Credentials](../getting-started/gs-credentials). |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:76]: `clientApiKey`

File: docs/sdks/uid2-sdk-ref-java.md [hunk 6]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
 
-| Parameter | Description | Recommended Value |
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the UID2 service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to UID2, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+| Parameter | Description | 
+| :--- | :--- |
+| `endpoint` | The endpoint for the UID2 service. See [Environments](../getting-started/gs-environments) | 
+| `auth_key` | The API key. See [UID2 Credentials](../getting-started/gs-credentials). | 
+| `secret_key` | The client secret. See [UID2 Credentials](../getting-started/gs-credentials). |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:77]: `base64SecretKey`

File: docs/sdks/uid2-sdk-ref-java.md [hunk 7]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+| `NOT_AUTHORIZED_FOR_MASTER_KEY` | The requester does not have authorization to use the master key.       |
+| `NOT_INITIALIZED`               | The client library is waiting to be initialized.                       |
+| `KEYS_NOT_SYNCED`               | The client has failed to synchronize keys from the UID2 service.       |
+| `ENCRYPTION_FAILURE`            | A generic encryption failure occurred.                                 |
+
+### Decryption Response Content
+
+Whether decrypting with the `BidstreamClient` or the `SharingClient`, the SDK returns the information shown in the following table.
+
+| Property      | Description                                                                                                                                     |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:116]: Really we have methods `getStatus()` etc

File: docs/sdks/uid2-sdk-ref-java.md [hunk 8]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
 :::note
 When you use an SDK, you do not need to store or manage decryption keys.
 :::
 
-If you're a DSP, for bidding, call the interface to decrypt a UID2 advertising token and return the UID2. For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
+### Encryption Response Content
+
+When encrypting with the `SharingClient`, the SDK returns the information shown in the following table.
+
+| Property         | Description                                                                                                                                     |
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:94]: Same comment as below, we have methods not properties

File: docs/sdks/uid2-sdk-ref-java.md [hunk 9]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
-| `GetUid()` | The raw UID2 for the corresponding UID2 advertising token. |
-| `GetEstablished()` | The timestamp indicating when a user first established the UID2 with the publisher. |
+3. Decrypt a token into a raw UID2. Pass the domain name of the site where the bid originated from:
+
+```java
+DecryptionResponse decrypted = client.decryptTokenIntoRawUid(uidToken, domain); 
+//If decryption succeeded, use the raw UID2.
+if (decrypted.isSuccess()) 
+{
+    //Use decrypted.uid
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:161]: decrypted.getUid()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 10]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+}
+else 
+{
+    // Check decrypted.status for the failure reason.
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:165]: getStatus()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 11]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+client.refresh();
+```
+
+3. If you are a sender, call `encryptRawUidIntoToken`:
+```java
+EncryptionDataResponse encrypted = client.encryptRawUidIntoToken(raw_uid);
+// If encryption succeeded, send the UID2 token to the receiver.
+if (encrypted.isSuccess())
+{
+        // Send encrypted.uid to receiver
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:197]: getUid()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 12]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+}
+else
+{
+        // Check encrypted.status for the failure reason.
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:201]: getStatus()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 13]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+}
+```
+If you are a receiver, call `decryptTokenIntoRawUid`:
+
+```java
+DecryptionResponse decrypted = client.decryptTokenIntoRawUid(uid_token);
+// If decryption succeeded, use the raw UID2.
+if (decrypted.isSuccess())
+{
+    //  Use decrypted.uid
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:211]: getUid()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 14]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+}
+else
+{
+    // Check decrypted.status for the failure reason.
```
Comment (mcollins-ttd) [inline docs/sdks/uid2-sdk-ref-java.md:215]: getStatus()

File: docs/sdks/uid2-sdk-ref-java.md [hunk 15]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
-| Value | Description |
-| :--- | :--- |
-| `Success` | The UID2 advertising token was decrypted successfully and a raw UID2 was returned. |
-| `NotAuthorizedForKey` | The requester does not have authorization to decrypt this UID2 advertising token.|
-| `NotInitialized` | The client library is waiting to be initialized. |
-| `InvalidPayload` | The incoming UID2 advertising token is not a valid payload. |
-| `ExpiredToken` | The incoming UID2 advertising token has expired. |
-| `KeysNotSynced` | The client has failed to synchronize keys from the UID2 service. |
-| `VersionNotSupported` |  The client library does not support the version of the encrypted token. |
+## Usage for UID2 Sharers
```
Comment (sunnywu) [inline docs/sdks/uid2-sdk-ref-java.md:171]: move this subject to the decision made in the Python SDK PR [here](https://github.com/IABTechLab/uid2docs/pull/511/files#r1560460611)

File: docs/sdks/uid2-sdk-ref-java.md [hunk 16]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+| Value                      | Description                                                             |
+|:---------------------------|:------------------------------------------------------------------------|
+| `SUCCESS`                  | The UID2 token was decrypted successfully and a raw UID2 was returned.  |
+| `NOT_AUTHORIZED_FOR_KEY`   | The requester does not have authorization to decrypt this UID2 token.   |
+| `NOT_INITIALIZED`          | The client library is waiting to be initialized.                        |
+| `INVALID_PAYLOAD`          | The incoming UID2 token is not a valid payload.                         |
+| `EXPIRED_TOKEN`            | The incoming UID2 token has expired.                                    |
+| `KEYS_NOT_SYNCED`          | The client has failed to synchronize keys from the UID2 service.        |
+| `VERSION_NOT_SUPPORTED`    | The client library does not support the version of the encrypted token. |
+| `INVALID_TOKEN_LIFETIME`   | The token has invalid timestamp.        |
```
Comment (genwhittTTD) [inline docs/sdks/uid2-sdk-ref-java.md:135]: The token has an invalid timestamp.
(just add "an")

File: docs/sdks/uid2-sdk-ref-java.md [hunk 17]
```diff
@@ -63,56 +58,165 @@ The binary is published on the Maven repository:
...
+
+## Usage for DSPs
+
+
+The following instructions provide an example of how you can decode bid stream tokens using the UID2 SDK for Java as a DSP.
```
Comment (genwhittTTD) [inline docs/sdks/uid2-sdk-ref-java.md:140]: Suggest turn it around:

The following instructions provide an example of how a DSP can decode bid stream tokens using the UID2 SDK for Java.
or
The following instructions provide an example of how you, as a DSP, can decode bid stream tokens using the UID2 SDK for Java.

**Review by genwhittTTD — COMMENTED** [review]
Couple of v small copy edits marked.

File: docs/sdks/uid2-sdk-ref-java.md (hunk 4 above)
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:75]: Yeah, that makes sense.

File: docs/sdks/uid2-sdk-ref-java.md (hunk 2 above)
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:154]: Yeah, that makes sense.

File: docs/sdks/uid2-sdk-ref-java.md (hunk 1 above)
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:69]: My bad, good catch!

Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:69]: My bad, good catch! Thank you.

File: docs/sdks/uid2-sdk-ref-java.md (hunk 8 above)
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:94]: Yeah, originally I thought that was only comments. I should make that more accurate.

File: docs/sdks/uid2-sdk-ref-java.md [hunk 18]
```diff
@@ -63,56 +58,116 @@ The binary is published on the Maven repository:
...
+```java
+client.refresh();
+```
 
-| Function | Description |
-| :--- | :--- |
-| `GetStatus()` | The decryption result status. For a list of possible values and definitions, see [Response Statuses](#response-statuses). |
-| `GetUid()` | The raw UID2 for the corresponding UID2 advertising token. |
-| `GetEstablished()` | The timestamp indicating when a user first established the UID2 with the publisher. |
+3. Decrypt a token into a raw UID2. Pass the token, and the domain name of the site where the bid originated from. The domain name must be all lower case, without spaces and without subdomain. For example, for `Subdomain.DOMAIN.com` , pass `domain.com` instead:
```
Comment (genwhittTTD) [inline docs/sdks/uid2-sdk-ref-java.md:155]: Remove extra space in this:
For example, for `Subdomain.DOMAIN.com` ,
Updated line:
3. Decrypt a token into a raw UID2. Pass the token, and the domain name of the site where the bid originated from. The domain name must be all lower case, without spaces and without subdomain. For example, for `Subdomain.DOMAIN.com`, pass `domain.com` instead:

File: docs/sdks/uid2-sdk-ref-java.md [hunk 19]
```diff
@@ -63,56 +58,116 @@ The binary is published on the Maven repository:
...
+| `NOT_AUTHORIZED_FOR_KEY`        | The requester does not have authorization to use the encryption key.   |
+| `NOT_AUTHORIZED_FOR_MASTER_KEY` | The requester does not have authorization to use the master key.       |
+| `NOT_INITIALIZED`               | The client library is waiting to be initialized.                       |
+| `KEYS_NOT_SYNCED`               | The client has failed to synchronize keys from the UID2 service.       |
+| `ENCRYPTION_FAILURE`            | A generic encryption failure occurred.                                 |
+
+### Decryption Response Content
 
-The following is the decrypt method in Java:
+Whether decrypting with the `BidstreamClient` or the `SharingClient`, the SDK returns the information shown in the following table.
```
[the remark this answers is not in the document]
Comment (jon8787) [reply docs/sdks/uid2-sdk-ref-java.md:115]: probably came from https://unifiedid.com/docs/sdks/uid2-sdk-ref-csharp-dotnet

File: docs/sdks/uid2-sdk-ref-java.md [hunk 20]
```diff
@@ -63,56 +58,116 @@ The binary is published on the Maven repository:
...
+The `SharingClient` class allows you to encrypt raw UID2s into UID2 tokens and decrypt UID2 tokens into raw UID2s.
 
 :::note
 When you use an SDK, you do not need to store or manage decryption keys.
 :::
 
-If you're a DSP, for bidding, call the interface to decrypt a UID2 advertising token and return the UID2. For details on the bidding logic for handling user opt-outs, see [DSP Integration Guide](../guides/dsp-guide.md).
+### Encryption Response Content
+
+When encrypting with the `SharingClient`, the SDK returns the information shown in the following table.
```
[the remark this answers is not in the document]
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:93]: Thank you, it is more clear. will change.

File: docs/sdks/uid2-sdk-ref-java.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:115]: Will change, it is more clear. Yeah, it is from https://unifiedid.com/docs/sdks/uid2-sdk-ref-csharp-dotnet

File: docs/sdks/uid2-sdk-ref-java.md [hunk 21]
```diff
@@ -63,56 +58,116 @@ The binary is published on the Maven repository:
...
+## Initialization
 
-The initialization function configures the parameters necessary for the SDK to authenticate with the UID2 service. It also allows you to configure retry intervals in the event of errors.
+The initialization step depends on the role, as shown in the following table.
 
-| Parameter | Description | Recommended Value |
-| :--- | :--- | :--- |
-| `endpoint` | The endpoint for the UID2 service. | N/A |
-| `authKey` | The authentication token that belongs to the client. For access to UID2, see [Contact Info](../getting-started/gs-account-setup.md#contact-info). | N/A |
+| Role        | Create Instance of Class | Link to Instructions                              |
```
[the remark this answers is not in the document]
Comment (caroline-ttd) [reply docs/sdks/uid2-sdk-ref-java.md:65]: Yeah, that is a good catch, thank you.

**Review by genwhittTTD — APPROVED** [review]
LGTM thx! Apologies for the delay.

**Review by jon8787 — APPROVED** [review]

**Review by genwhittTTD — APPROVED** [review]
LGTM thx!

**Pull request merged**